Examples/ObjectiveC/CacheManagementExample.m

```diff
@@ -0,0 +1,158 @@
+#import "CacheManagementExample.h"
+@import Mapbox;
+
+NSString *const MBXExampleCacheManagement = @"CacheManagementExample";
+
+@interface CacheManagementExample () <MGLMapViewDelegate>
+
+@property (nonatomic) MGLMapView *mapView;
+@property (nonatomic) UIAlertController *alertController;
+@end
+
+@implementation CacheManagementExample
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    /* Set the maximum ambient cache size in bytes. Call this method before the map view is loaded.
+
+     The ambient cache is created through the end user loading and using a map view. */
+    NSUInteger maxCacheSize = 62914560;
+    [[MGLOfflineStorage sharedOfflineStorage] setMaximumAmbientCacheSize:maxCacheSize withCompletionHandler:^(NSError * _Nullable error) {
+        if (error != nil) {
+            NSLog(@"Unable to set maximum ambient cache size: %@", error.localizedDescription);
+        }
+
+        self.mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds];
```

Comment:
Creating the map view inside this block strikes me as precarious/unusual — it forces you to unnecessarily consider threading and object retention.

Reply:
👍  I need to call the method before the map is loaded. `setMaximumAmbientCacheSize` is called on the main thread, but I can move the method back to AppDelegate.

Reply:
Let’s keep the cache stuff contained to this example and perhaps make a note that the method might be better called elsewhere.

My suggestion for how to do the map init would be to extract it into a different method.